frontend/src/Components/StudentMetrics.tsx

```diff
@@ -0,0 +1,197 @@
+                                    {(
+                                        (metrics.active_users /
+                                            metrics.total_users) *
+                                        100
+                                    ).toFixed(2)}
+                                    %
+                                </div>
+                            </div>
+                            <div className="stat">
+                                <div className="stat-title">Total Logins</div>
+                                <div className="stat-value">
+                                    {metrics.total_logins}
+                                </div>
+                            </div>
+                            <div className="stat">
+                                <div className="stat-title">Logins per Day</div>
+                                <div className="stat-value">
+                                    {metrics.logins_per_day}
+                                </div>
+                            </div>
+                            <div className="stat">
+                                <div className="stat-title">
+                                    New Residents Added
+                                </div>
+                                <div className="stat-value">
+                                    {metrics.new_residents_added}
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+
+                    {/* Peak Login Times Chart */}
+                    <div className="mb-6">
+                        <h3 className="text-xl font-semibold mb-2">
+                            Peak Login Times
+                        </h3>
+                        <div className="flex flex-row gap-6">
+                            <BarChart
+                                width={600}
+                                height={300}
+                                data={metrics?.peak_login_times}
+                            >
+                                <CartesianGrid strokeDasharray="3 3" />
+                                <XAxis
+                                    dataKey="time_interval"
+                                    tickFormatter={(tick: string) =>
+                                        new Date(tick).toLocaleTimeString([], {
+                                            hour: '2-digit',
+                                            minute: '2-digit'
+                                        })
+                                    }
+                                />
+                                <YAxis allowDecimals={false} />
+                                <Tooltip />
+                                <Legend />
+                                <Bar dataKey="total_logins" fill="#8884d8" />
+                            </BarChart>
+                        </div>
+                    </div>
```

Comment:
typically i have been extracting charts into their own component that way we can reuse them in other pages if we need to in the future

Reply:
Also, try to use colors we already use in the application rather than this purple `#8884d8`